src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextDocumentCreatedListener.cs

```diff
@@ -207,5 +224,20 @@ private void InitializeOptions(ITextBuffer textBuffer)
                 razorOptions.SetOptionValue(DefaultOptions.TabSizeOptionId, (int)tabSize);
             }
         }
+
+        private void RazorGuestBuffer_Changed(object sender, TextContentChangedEventArgs args)
+        {
+            var replacePlaceholderChange = args.Changes.FirstOrDefault(c => c.OldText == LanguageServerConstants.CursorPlaceholderString && c.NewText == string.Empty);
```

Comment:
HAH! The guest actually sees these changes??? That's a dope hack fix 👏

Reply:
Yup. Even though we do it all as part of the same undo transaction, we apply these text changes separately. So we see them on the guest.